### DISCUSSION TITLE
JQ syntax editor/plugin for jetbrains products

### DISCUSSION THREAD

**Discussion by chb0github** [stedolan/jq#3045] [Ideas]
This is a feature request for support and integration with jetbrains editorsfor jq syntax. Jetbrains offers a plugin framework for such things, and it would be a great way to further support the jq community:

Specifically:
1. Syntax highlighting
2. syntax error/validation
3. debug

Perhaps a new repo for the effort, so it can grow independently. I recognize this doesn't specifically align to this repo, but I saw no where else to publish the request

**Comment by wader** [comment]
If it supports LSP there is https://github.com/wader/jq-lsp which i use for https://github.com/wader/vscode-jq which also has example of grammar for syntax highlighting.

**Reply by kishaningithub** [reply]
Can this be brought under the umbrella of [jqlang org](https://github.com/jqlang) ?

**Reply by wader** [reply]
Yeap might be good idea. I've thought about if https://github.com/wader/jqjq should be move there also. But then i guess it should be seen as more "official"? not sure what other maintainers think?

jq-lsp started as quick hack that happen to work weridly well, so unfortunately the code is a mess atm and hard to work it. Ideally i think it would make more sense to rewrite it using tree-sitter etc. I wonder how hard it would be to plumb together tree-sitter with a jq implementation to still implement the language sever part in jq?

**Comment by nicowilliams** [comment]
> If it supports LSP there is https://github.com/wader/jq-lsp which i use for https://github.com/wader/vscode-jq which also has example of grammar for syntax highlighting.

My goodness, you wrote a language server for jq?!  That's awesome!

**Reply by blurayne** [reply]
I guess it was more targeted at fq ;)

**Comment by nicowilliams** [comment]
> If it supports LSP there is https://github.com/wader/jq-lsp which i use for https://github.com/wader/vscode-jq which also has example of grammar for syntax highlighting.

We should link this from the README!

**Comment by wader** [comment]
Currently the installation of both jq-lsp and vscode-jq is not very polished, no packing and i'm not sure it even works on windows :). Maybe we could for now add some section to the wiki with development support stuff?

btw have you seen https://github.com/fiatjaf/awesome-jq? maybe could cherry pick some stuff from there also

**Comment by trothwell** [comment]
I was able to get file type association working with this plugin: https://github.com/redhat-developer/lsp4ij

Unfortunately, I'm unable to get JqJSON listed as a "language" so other plugins that have nested language support won't work.

**Comment by wader** [comment]
@trothwell nice, if it works well i'm happy to add it to jq-lsp:s README.md

**Comment by angelozerr** [comment]
@trothwell I suggest that you create a PR to LSP4IJ to add it your config as LSP template. Please read https://github.com/redhat-developer/lsp4ij/blob/main/docs/UserDefinedLanguageServerTemplate.md#how-to-contribute-a-user-defined-language-server-template

I don't know how to install jq-lsp, but you can write also an installer.json to install it automaticly.

If someone explains me:

 * the command which starts jq-lsp
 *  how to install jq-lsp (should we install some runtime, is it possible to do that with terminal command), I could create a quick PR in LSP4IJ.

**Reply by wader** [reply]
Hello, i'm the author of jq-lsp. At moment there are not pre-built binaries. Most LSP-setup tools/frameworks seems to build and install things on their own. For jq-lsp that would mean using `go install` etc, see https://github.com/wader/jq-lsp?tab=readme-ov-file#install

**Reply by angelozerr** [reply]
Thanks @wader ! Coudl you give me the start command to start jq-lsp please.

**Reply by wader** [reply]
If you start it with no arguments it will go into "LSP-mode", that is talk the LSP-protocol over stdin/stdout.

**Reply by wader** [reply]
So for lsp4ilj i guess just something like this if jq-lsp is in `$PATH`
```
  "programArgs": {
    "default": "jq-slp"
  },
```

let me know how it goes, it would not surprise if something acts weird so we need to tweak something

Note that i haven't tried jq-lsp with windows at all yet.

**Reply by angelozerr** [reply]
Indeed I have tried that and it is working. I will soon push my work.

**Reply by wader** [reply]
and feel free to do a jq-lsp PR to add it to README.md

**Reply by angelozerr** [reply]
See doc at https://github.com/redhat-developer/lsp4ij/blob/main/docs/user-defined-ls/jq-lsp.md

**Reply by wader** [reply]
Nice looking docs 👍
